src/hypothesis/internal/compat.py

```diff
@@ -501,11 +501,17 @@ def implements_iterator(it):
 
 
 if PY2:
+    # We use integer operations because math.floor etc return floats, which
+    # lose odd numbers at large values.  See tests/py2/test_ceil_floor.py
```

Comment:
IEEE double-precision floats (the Python float type) can represent all integers up to 2^52 + 2, but only some thereafter.  This variable precision is why they're called floating-point!

I'll clarify the comments and add the example that `float(2**53) == float(2**53 + 1)` instead of a reference to the tests.